### PR TITLE
Add conditions for implicit framework references for .NET 3.5 and below

### LINKED ISSUE #532: Targeting net20 results in broken reference warnings
_From @jnm2 on December 17, 2016 17:54_

(I'm multitargeting `net20` and `netstandard1.3` but this happens also when just targeting `net20`.)

Under Dependencies > Assemblies I have `System`, `System.Core`, `System.Data`, `System.Drawing`, `System.Runtime.Serialization`, `System.Xml` and `System.Xml.Linq`. Of these I only actually need `System`, so ordinarily I'd delete the rest. However deleting is impossible since they are defined outside my .csproj.

Not being able to delete doesn't bother me too much, except that I get build warnings for `System.Core`, `System.Runtime.Serialization` and `System.Xml.Linq` due to targeting `net20`.

If you are going to require mandatory assembly references if one of the targets is .NET Framework, I'd expect you to detect which references make sense to add given the framework version. (Personally I'd be even more happy with no mandatory references and having to list them in the .csproj so I can delete all of them except `System`.)

_Copied from original issue: dotnet/roslyn-project-system#987_

### PR DESCRIPTION
**Pull request by dsplaisted**
Fixes #532 

There aren't any tests for this, as we run all our tests using the .NET CLI version of MSBuild, which doesn't support targeting .NET 3.5 and lower (as there are no reference assemblies for these frameworks).

### CODE REVIEW

**Review by srivatsn — APPROVED** [review]

File: src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.BeforeCommon.targets [hunk 1]
```diff
@@ -77,18 +77,18 @@ Copyright (c) .NET Foundation. All rights reserved.
   </PropertyGroup>
   
   <ItemGroup Condition=" '$(DisableImplicitFrameworkReferences)' != 'true' and '$(TargetFrameworkIdentifier)' == '.NETFramework'">
+    <!-- When doing greater than/less than comparisons between strings, MSBuild will try to parse the strings as Version objects and compare them as
```
Comment (davidfowl) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.BeforeCommon.targets]: Separate item groups with conditions would be clearer.

Comment (dsplaisted) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.BeforeCommon.targets]: There's already a condition on the item group: `Condition=" '$(DisableImplicitFrameworkReferences)' != 'true' and '$(TargetFrameworkIdentifier)' == '.NETFramework'"`

So splitting into separate item groups based on the target framework version would mean duplicating that condition and adding an additional clause for the version.  I don't think that would end up being clearer.

Comment (davidfowl) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.BeforeCommon.targets]: Then manually group by similar conditions?

Comment (dsplaisted) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.BeforeCommon.targets]: OK, Done

**Comment by onovotny** [comment]
Shouldn't a similar set of changes be included for the Xamarin platforms? They need/should have implicit references added as well to match, plus their platform lib `Xamarin.iOS`, `MonoAndroid`, etc?

**Review by natidea — APPROVED** [review]

**Comment by dsplaisted** [comment]
@onovotny 

> Shouldn't a similar set of changes be included for the Xamarin platforms? They need/should have implicit references added as well to match, plus their platform lib `Xamarin.iOS`, `MonoAndroid`, etc?

I've added a comment on #491 to do this.

**Pull request merged**